app/controllers/application_controller.rb

```diff
@@ -19,6 +19,8 @@ class ApplicationController < ActionController::Base
   before_action :update_persistent_announcements
   before_action :set_breadcrumbs
 
+  helper_method :resource_name, :resource_class, :resource, :devise_mapping
```

Comment:
Are these being used by devise and if so, do they need to have these names?  If that's the case, maybe leave a comment explaining that.